stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js

```diff
@@ -0,0 +1,171 @@
+/* global testRule */
+
+const {getCustomPropertyNames, tokens} = require('@shopify/polaris-tokens');
+
+const {messages, ruleName} = require('.');
+
+const polarisCustomPropertyNames = getCustomPropertyNames(tokens);
+
+// For external usage:
+// -p-* Tokens are reserved for usage by polaris-tokens.
+//   - Defining custom properties with this prefix is disallowed
+//   - Usage of them is allowed
+// -pc-* Tokens are reserved for usage by components in polaris-react
+//  - Defining custom properties with this prefix is disallowed
+//  - Usage of them is disallowed
+// Any other tokens can be freely defined and used
+const externalUserDefinedCustomPropertyNames = /--(?!pc?-).+/;
+const externalConfig = [
+  {
+    allowedProperties: [externalUserDefinedCustomPropertyNames],
+    allowedValues: {
+      '/.+/': [
+        ...polarisCustomPropertyNames,
+        externalUserDefinedCustomPropertyNames,
+      ],
+    },
+  },
+];
+
+// For internal usage:
+// -p-* Tokens are reserved for usage by polaris-tokens.
+//   - Defining custom properties with this prefix is disallowed (as they get
+//     generated by the CustomProperties component)
+//   - Usage of them is allowed
+// -pc-* Tokens are reserved for usage by components in polaris-react
+//  - Defining custom properties with this prefix is allowed
+//  - Usage of them is allowed
+// Any other tokens should not be defined or used
+const internalPolarisComponentCustomProperties = /--pc-.+/;
+const internalConfig = [
+  {
+    allowedProperties: [
+      '--polaris-version-number',
+      internalPolarisComponentCustomProperties,
+    ],
+    allowedValues: {
+      '/.+/': [
+        internalPolarisComponentCustomProperties,
+        ...polarisCustomPropertyNames,
+      ],
+    },
+  },
+];
+
+testRule({
+  ruleName,
+  plugins: [__dirname],
+  config: externalConfig,
+  accept: [
+    {
+      code: '.a { border: 1px red; }',
+      description: 'Uses no custom properties',
+    },
+    {
+      code: '.a { --test: red; }',
+      description:
+        "Defining custom-properties that don't start with --p- or --pc-",
+    },
+    {
+      code: '.a { color: --p-text; }',
+      description: 'Using custom-properties from polaris-tokens',
+    },
+    {
+      code: '.a { --test: red; color: --test; }',
+      description: 'Using other custom-properties',
+    },
+  ],
+
+  reject: [
+    {
+      code: '.a { --p-test: red; }',
+      description:
+        'Defining custom-properties that start with --p- is disallowed',
+      message: messages.rejected('--p-test'),
+      line: 1,
+      column: 6,
+      endLine: 1,
+      endColumn: 20,
+    },
+    {
+      code: '.a { --pc-test: red; }',
+      description:
+        'Defining custom-properties that start with --pc- is disallowed',
+      message: messages.rejected('--pc-test'),
+      line: 1,
+      column: 6,
+      endLine: 1,
+      endColumn: 21,
+    },
+    {
+      code: '.a { color: --p-absent; }',
```

Comment:
```suggestion
      code: '.a { color: var(--p-unknown); }',
```

Included light suggestion to use `unknown` to indicate usage of a non-polaris-token

Reply:
Updated, thanks!